### PR TITLE
Update `ImplicitFreeSurface` & `HeptadiagonalIterativeSolver` docstrings

### LINKED ISSUE #2620: Need to document the available solver method options for ImplicitFreeSurface solvers
https://github.com/CliMA/Oceananigans.jl/blob/65b8851d37052e90ca4a3e0c4a1c20398b0ee09a/src/Models/HydrostaticFreeSurfaceModels/implicit_free_surface.jl#L35-L58

cc: @elise-palethorpe

### PR DESCRIPTION
**Pull request by navidcy**
Enhances docstring for the `ImplicitFreeSurface` constructor to include the available solver methods.

Closes #2620

### CODE REVIEW

**Review by francispoulin — APPROVED** [review]

File: src/Solvers/heptadiagonal_iterative_solver.jl [hunk 1]
```diff
@@ -60,32 +61,30 @@ Axᵢ₊₁ ηᵢ₊₁ + Axᵢ ηᵢ₋₁ + Ayⱼ₊₁ ηⱼ₊₁ + Ayⱼ η
...
-(`matrix_from_coefficients`).
+to [`matrix_from_coefficients`](@ref).
 
-The diagonal term `- Az / (g * Δt²)` is added later on during the time stepping
-to allow for variable time step. It is updated only when the previous time step 
-is different (`Δt_previous != Δt`).
+To allow for variable time step, the diagonal term `- Az / (g * Δt²)` is only added later on
+and it is updated only when the previous time step changes (`previous_Δt != Δt`).
 
 Preconditioning is done through the incomplete LU factorization.
```
Comment (simone-silvestri) [inline src/Solvers/heptadiagonal_iterative_solver.jl:76]:
```suggestion
Preconditioning is done through the various methods implemented in `sparse_preconditioners.jl`. 
```

File: src/Solvers/heptadiagonal_iterative_solver.jl [hunk 2]
```diff
@@ -60,32 +61,30 @@ Axᵢ₊₁ ηᵢ₊₁ + Axᵢ ηᵢ₋₁ + Ayⱼ₊₁ ηⱼ₊₁ + Ayⱼ η
...
-to allow for variable time step. It is updated only when the previous time step 
-is different (`Δt_previous != Δt`).
+To allow for variable time step, the diagonal term `- Az / (g * Δt²)` is only added later on
+and it is updated only when the previous time step changes (`previous_Δt != Δt`).
 
 Preconditioning is done through the incomplete LU factorization. 
 
-It works for GPU, but it relies on serial backward and forward substitution which are very
+`HeptadiagonalIterativeSolver` works for GPU, but it relies on serial backward and forward substitution which are very
 heavy and destroy all the computational advantage, therefore it is switched off until a
```
Comment (simone-silvestri) [inline src/Solvers/heptadiagonal_iterative_solver.jl:79]:
```suggestion
```

File: src/Solvers/heptadiagonal_iterative_solver.jl [hunk 3]
```diff
@@ -60,32 +61,30 @@ Axᵢ₊₁ ηᵢ₊₁ + Axᵢ ηᵢ₋₁ + Ayⱼ₊₁ ηⱼ₊₁ + Ayⱼ η
...
 parallel backward/forward substitution is implemented. It is also updated based on the
```
Comment (simone-silvestri) [inline src/Solvers/heptadiagonal_iterative_solver.jl:80]:
```suggestion
```

File: src/Solvers/heptadiagonal_iterative_solver.jl [hunk 4]
```diff
@@ -60,32 +61,30 @@ Axᵢ₊₁ ηᵢ₊₁ + Axᵢ ηᵢ₋₁ + Ayⱼ₊₁ ηⱼ₊₁ + Ayⱼ η
...
-matrix when `Δt != Δt_previous`
+matrix when `previous_Δt != Δt`.
```
Comment (simone-silvestri) [inline src/Solvers/heptadiagonal_iterative_solver.jl:81]:
```suggestion
```

**Comment by simone-silvestri** [comment]
I was trying to do a multiline suggestion and I kind of messed up

**Pull request merged**